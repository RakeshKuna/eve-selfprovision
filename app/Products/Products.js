/**
 *
 * App
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 */

import React from 'react';
import { ETable } from 'containers/ETable';
import { Link } from 'react-router-dom';
import { Button, Row, Col } from 'reactstrap';
import userService from 'services';
import { history } from 'helpers';

export class Products extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: [],
      loading: false
    };
    this.columns = [{
      Header: 'Currency Code',
      accessor: 'currencyCode'
    }, {
      Header: 'Product Description',
      accessor: 'productDescription'
    }, {
      Header: 'Product Category',
      accessor: 'productCategory'
    }, {
      Header: 'Channel Partner Description',
      accessor: 'channelPartnerDescription'
    }, {
      Header: 'Renewable',
      accessor: 'renewable'
    }, {
      Header: 'Service Type',
      accessor: 'serviceType'
    }, {
      Header: 'Period',
      accessor: 'period'
    }, {
      Header: 'Basic Service',
      accessor: 'basicService'
    }, {
      Header: 'SKU OR Quick Code',
      accessor: 'skuORQuickCode'
    }, {
      Header: 'DMA Name',
      accessor: 'dmaName'
    }, {
      Header: 'Retail Price',
      accessor: 'retailPrice'
    }, {
      Header: 'Duration',
      accessor: 'duration'
    }, {
      Header: 'Display Order',
      accessor: 'displayOrder'
    }, {
      Header: 'is Ads Enabled',
      accessor: 'isAdsEnabled'
    }, {
      Header: 'Display Name',
      accessor: 'displayName'
    }, {
      Header: 'Product Name',
      accessor: 'productName'
    }, {
      Header: 'Actions',
      accessor: '[row identifier to be passed to button]',
      Cell: (row) => (<Button color="primary" onClick={() => this.editProducts(row.original)} className="view_btn">Edit</Button>),
      filterable: false
    }];
  }

     getProducts=() => {
       this.setState({
         loading: true
       });
       userService.getProductsList(this.props.channelPartnerId)
         .then(
           (success) => {
             if (!Array.isArray(success.productsResponseMessage)) {
               success.productsResponseMessage = [success.productsResponseMessage];
             }
             this.setState({
               data: success.productsResponseMessage,
               loading: false
             });

             const skus = [];
             success.productsResponseMessage.forEach((element) => {
               skus.push(element.skuORQuickCode);
             });
             this.props.setSku(skus);
           },
           () => {
             this.setState({
               loading: false
             });
           }
         );
     }
     editProducts=(product) => {
       history.push({ pathname: `/${this.props.channelPartnerId}/updateProduct/${product.productName}`, state: { product } });
     }
     render() {
       return (
         <div>
           <Row className="mb-4">
             <Col sm={8}>
               <h2>Products</h2>
             </Col>
             <Col sm={4} className="text-right">
               <Button color="primary" className="mr-2" tag={Link} to={`/businessUnit/${this.props.channelPartnerId}/addProduct`}>ADD</Button>
             </Col>
           </Row>
           <Row>
             <Col>
               <ETable key="A" data={this.state.data} columns={this.columns} loading={this.state.loading} />
             </Col>
           </Row>
         </div>
       );
     }
}
export default Products;
