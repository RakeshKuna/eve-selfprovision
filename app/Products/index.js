export { default as Products } from './Products';
export { default as AddProducts } from './AddProducts';
