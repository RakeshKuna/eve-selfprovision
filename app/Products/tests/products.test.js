import React from 'react';
import { shallow } from 'enzyme';
import { Products } from '../Products';

describe('Products', () => {
  it('<Row> element should render', () => {
    const wrapper = shallow(<Products />);
    expect(wrapper).toMatchSnapshot();
  });
  it('<Row> on Button redirect', () => {
    const wrapper = shallow(<Products />);
    wrapper.find('Button').at(0).props('to');
  });
});
