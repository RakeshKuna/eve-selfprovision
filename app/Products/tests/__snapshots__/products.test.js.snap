// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products <Row> element should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Products />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Row
          className="mb-4"
          tag="div"
        >
          <Col
            sm={8}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <h2>
              Products
            </h2>
          </Col>
          <Col
            className="text-right"
            sm={4}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Button
              className="mr-2"
              color="primary"
              tag={[Function]}
              to="/businessUnit/undefined/addProduct"
            >
              ADD
            </Button>
          </Col>
        </Row>,
        <Row
          tag="div"
        >
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Connect(ETable)
              columns={
                Array [
                  Object {
                    "Header": "Currency Code",
                    "accessor": "currencyCode",
                  },
                  Object {
                    "Header": "Product Description",
                    "accessor": "productDescription",
                  },
                  Object {
                    "Header": "Product Category",
                    "accessor": "productCategory",
                  },
                  Object {
                    "Header": "Channel Partner Description",
                    "accessor": "channelPartnerDescription",
                  },
                  Object {
                    "Header": "Renewable",
                    "accessor": "renewable",
                  },
                  Object {
                    "Header": "Service Type",
                    "accessor": "serviceType",
                  },
                  Object {
                    "Header": "Period",
                    "accessor": "period",
                  },
                  Object {
                    "Header": "Basic Service",
                    "accessor": "basicService",
                  },
                  Object {
                    "Header": "SKU OR Quick Code",
                    "accessor": "skuORQuickCode",
                  },
                  Object {
                    "Header": "DMA Name",
                    "accessor": "dmaName",
                  },
                  Object {
                    "Header": "Retail Price",
                    "accessor": "retailPrice",
                  },
                  Object {
                    "Header": "Duration",
                    "accessor": "duration",
                  },
                  Object {
                    "Header": "Display Order",
                    "accessor": "displayOrder",
                  },
                  Object {
                    "Header": "is Ads Enabled",
                    "accessor": "isAdsEnabled",
                  },
                  Object {
                    "Header": "Display Name",
                    "accessor": "displayName",
                  },
                  Object {
                    "Header": "Product Name",
                    "accessor": "productName",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Actions",
                    "accessor": "[row identifier to be passed to button]",
                    "filterable": false,
                  },
                ]
              }
              data={Array []}
              loading={false}
            />
          </Col>
        </Row>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Col
              sm={8}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <h2>
                Products
              </h2>
            </Col>,
            <Col
              className="text-right"
              sm={4}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Button
                className="mr-2"
                color="primary"
                tag={[Function]}
                to="/businessUnit/undefined/addProduct"
              >
                ADD
              </Button>
            </Col>,
          ],
          "className": "mb-4",
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h2>
                Products
              </h2>,
              "sm": 8,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Products",
              },
              "ref": null,
              "rendered": "Products",
              "type": "h2",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                className="mr-2"
                color="primary"
                tag={[Function]}
                to="/businessUnit/undefined/addProduct"
              >
                ADD
              </Button>,
              "className": "text-right",
              "sm": 4,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "ADD",
                "className": "mr-2",
                "color": "primary",
                "tag": [Function],
                "to": "/businessUnit/undefined/addProduct",
              },
              "ref": null,
              "rendered": "ADD",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Connect(ETable)
              columns={
                Array [
                  Object {
                    "Header": "Currency Code",
                    "accessor": "currencyCode",
                  },
                  Object {
                    "Header": "Product Description",
                    "accessor": "productDescription",
                  },
                  Object {
                    "Header": "Product Category",
                    "accessor": "productCategory",
                  },
                  Object {
                    "Header": "Channel Partner Description",
                    "accessor": "channelPartnerDescription",
                  },
                  Object {
                    "Header": "Renewable",
                    "accessor": "renewable",
                  },
                  Object {
                    "Header": "Service Type",
                    "accessor": "serviceType",
                  },
                  Object {
                    "Header": "Period",
                    "accessor": "period",
                  },
                  Object {
                    "Header": "Basic Service",
                    "accessor": "basicService",
                  },
                  Object {
                    "Header": "SKU OR Quick Code",
                    "accessor": "skuORQuickCode",
                  },
                  Object {
                    "Header": "DMA Name",
                    "accessor": "dmaName",
                  },
                  Object {
                    "Header": "Retail Price",
                    "accessor": "retailPrice",
                  },
                  Object {
                    "Header": "Duration",
                    "accessor": "duration",
                  },
                  Object {
                    "Header": "Display Order",
                    "accessor": "displayOrder",
                  },
                  Object {
                    "Header": "is Ads Enabled",
                    "accessor": "isAdsEnabled",
                  },
                  Object {
                    "Header": "Display Name",
                    "accessor": "displayName",
                  },
                  Object {
                    "Header": "Product Name",
                    "accessor": "productName",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Actions",
                    "accessor": "[row identifier to be passed to button]",
                    "filterable": false,
                  },
                ]
              }
              data={Array []}
              loading={false}
            />
          </Col>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Connect(ETable)
              columns={
                Array [
                  Object {
                    "Header": "Currency Code",
                    "accessor": "currencyCode",
                  },
                  Object {
                    "Header": "Product Description",
                    "accessor": "productDescription",
                  },
                  Object {
                    "Header": "Product Category",
                    "accessor": "productCategory",
                  },
                  Object {
                    "Header": "Channel Partner Description",
                    "accessor": "channelPartnerDescription",
                  },
                  Object {
                    "Header": "Renewable",
                    "accessor": "renewable",
                  },
                  Object {
                    "Header": "Service Type",
                    "accessor": "serviceType",
                  },
                  Object {
                    "Header": "Period",
                    "accessor": "period",
                  },
                  Object {
                    "Header": "Basic Service",
                    "accessor": "basicService",
                  },
                  Object {
                    "Header": "SKU OR Quick Code",
                    "accessor": "skuORQuickCode",
                  },
                  Object {
                    "Header": "DMA Name",
                    "accessor": "dmaName",
                  },
                  Object {
                    "Header": "Retail Price",
                    "accessor": "retailPrice",
                  },
                  Object {
                    "Header": "Duration",
                    "accessor": "duration",
                  },
                  Object {
                    "Header": "Display Order",
                    "accessor": "displayOrder",
                  },
                  Object {
                    "Header": "is Ads Enabled",
                    "accessor": "isAdsEnabled",
                  },
                  Object {
                    "Header": "Display Name",
                    "accessor": "displayName",
                  },
                  Object {
                    "Header": "Product Name",
                    "accessor": "productName",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Actions",
                    "accessor": "[row identifier to be passed to button]",
                    "filterable": false,
                  },
                ]
              }
              data={Array []}
              loading={false}
            />,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "A",
            "nodeType": "class",
            "props": Object {
              "columns": Array [
                Object {
                  "Header": "Currency Code",
                  "accessor": "currencyCode",
                },
                Object {
                  "Header": "Product Description",
                  "accessor": "productDescription",
                },
                Object {
                  "Header": "Product Category",
                  "accessor": "productCategory",
                },
                Object {
                  "Header": "Channel Partner Description",
                  "accessor": "channelPartnerDescription",
                },
                Object {
                  "Header": "Renewable",
                  "accessor": "renewable",
                },
                Object {
                  "Header": "Service Type",
                  "accessor": "serviceType",
                },
                Object {
                  "Header": "Period",
                  "accessor": "period",
                },
                Object {
                  "Header": "Basic Service",
                  "accessor": "basicService",
                },
                Object {
                  "Header": "SKU OR Quick Code",
                  "accessor": "skuORQuickCode",
                },
                Object {
                  "Header": "DMA Name",
                  "accessor": "dmaName",
                },
                Object {
                  "Header": "Retail Price",
                  "accessor": "retailPrice",
                },
                Object {
                  "Header": "Duration",
                  "accessor": "duration",
                },
                Object {
                  "Header": "Display Order",
                  "accessor": "displayOrder",
                },
                Object {
                  "Header": "is Ads Enabled",
                  "accessor": "isAdsEnabled",
                },
                Object {
                  "Header": "Display Name",
                  "accessor": "displayName",
                },
                Object {
                  "Header": "Product Name",
                  "accessor": "productName",
                },
                Object {
                  "Cell": [Function],
                  "Header": "Actions",
                  "accessor": "[row identifier to be passed to button]",
                  "filterable": false,
                },
              ],
              "data": Array [],
              "loading": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Row
            className="mb-4"
            tag="div"
          >
            <Col
              sm={8}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <h2>
                Products
              </h2>
            </Col>
            <Col
              className="text-right"
              sm={4}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Button
                className="mr-2"
                color="primary"
                tag={[Function]}
                to="/businessUnit/undefined/addProduct"
              >
                ADD
              </Button>
            </Col>
          </Row>,
          <Row
            tag="div"
          >
            <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Connect(ETable)
                columns={
                  Array [
                    Object {
                      "Header": "Currency Code",
                      "accessor": "currencyCode",
                    },
                    Object {
                      "Header": "Product Description",
                      "accessor": "productDescription",
                    },
                    Object {
                      "Header": "Product Category",
                      "accessor": "productCategory",
                    },
                    Object {
                      "Header": "Channel Partner Description",
                      "accessor": "channelPartnerDescription",
                    },
                    Object {
                      "Header": "Renewable",
                      "accessor": "renewable",
                    },
                    Object {
                      "Header": "Service Type",
                      "accessor": "serviceType",
                    },
                    Object {
                      "Header": "Period",
                      "accessor": "period",
                    },
                    Object {
                      "Header": "Basic Service",
                      "accessor": "basicService",
                    },
                    Object {
                      "Header": "SKU OR Quick Code",
                      "accessor": "skuORQuickCode",
                    },
                    Object {
                      "Header": "DMA Name",
                      "accessor": "dmaName",
                    },
                    Object {
                      "Header": "Retail Price",
                      "accessor": "retailPrice",
                    },
                    Object {
                      "Header": "Duration",
                      "accessor": "duration",
                    },
                    Object {
                      "Header": "Display Order",
                      "accessor": "displayOrder",
                    },
                    Object {
                      "Header": "is Ads Enabled",
                      "accessor": "isAdsEnabled",
                    },
                    Object {
                      "Header": "Display Name",
                      "accessor": "displayName",
                    },
                    Object {
                      "Header": "Product Name",
                      "accessor": "productName",
                    },
                    Object {
                      "Cell": [Function],
                      "Header": "Actions",
                      "accessor": "[row identifier to be passed to button]",
                      "filterable": false,
                    },
                  ]
                }
                data={Array []}
                loading={false}
              />
            </Col>
          </Row>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Col
                sm={8}
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <h2>
                  Products
                </h2>
              </Col>,
              <Col
                className="text-right"
                sm={4}
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Button
                  className="mr-2"
                  color="primary"
                  tag={[Function]}
                  to="/businessUnit/undefined/addProduct"
                >
                  ADD
                </Button>
              </Col>,
            ],
            "className": "mb-4",
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <h2>
                  Products
                </h2>,
                "sm": 8,
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Products",
                },
                "ref": null,
                "rendered": "Products",
                "type": "h2",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Button
                  className="mr-2"
                  color="primary"
                  tag={[Function]}
                  to="/businessUnit/undefined/addProduct"
                >
                  ADD
                </Button>,
                "className": "text-right",
                "sm": 4,
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "ADD",
                  "className": "mr-2",
                  "color": "primary",
                  "tag": [Function],
                  "to": "/businessUnit/undefined/addProduct",
                },
                "ref": null,
                "rendered": "ADD",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Connect(ETable)
                columns={
                  Array [
                    Object {
                      "Header": "Currency Code",
                      "accessor": "currencyCode",
                    },
                    Object {
                      "Header": "Product Description",
                      "accessor": "productDescription",
                    },
                    Object {
                      "Header": "Product Category",
                      "accessor": "productCategory",
                    },
                    Object {
                      "Header": "Channel Partner Description",
                      "accessor": "channelPartnerDescription",
                    },
                    Object {
                      "Header": "Renewable",
                      "accessor": "renewable",
                    },
                    Object {
                      "Header": "Service Type",
                      "accessor": "serviceType",
                    },
                    Object {
                      "Header": "Period",
                      "accessor": "period",
                    },
                    Object {
                      "Header": "Basic Service",
                      "accessor": "basicService",
                    },
                    Object {
                      "Header": "SKU OR Quick Code",
                      "accessor": "skuORQuickCode",
                    },
                    Object {
                      "Header": "DMA Name",
                      "accessor": "dmaName",
                    },
                    Object {
                      "Header": "Retail Price",
                      "accessor": "retailPrice",
                    },
                    Object {
                      "Header": "Duration",
                      "accessor": "duration",
                    },
                    Object {
                      "Header": "Display Order",
                      "accessor": "displayOrder",
                    },
                    Object {
                      "Header": "is Ads Enabled",
                      "accessor": "isAdsEnabled",
                    },
                    Object {
                      "Header": "Display Name",
                      "accessor": "displayName",
                    },
                    Object {
                      "Header": "Product Name",
                      "accessor": "productName",
                    },
                    Object {
                      "Cell": [Function],
                      "Header": "Actions",
                      "accessor": "[row identifier to be passed to button]",
                      "filterable": false,
                    },
                  ]
                }
                data={Array []}
                loading={false}
              />
            </Col>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Connect(ETable)
                columns={
                  Array [
                    Object {
                      "Header": "Currency Code",
                      "accessor": "currencyCode",
                    },
                    Object {
                      "Header": "Product Description",
                      "accessor": "productDescription",
                    },
                    Object {
                      "Header": "Product Category",
                      "accessor": "productCategory",
                    },
                    Object {
                      "Header": "Channel Partner Description",
                      "accessor": "channelPartnerDescription",
                    },
                    Object {
                      "Header": "Renewable",
                      "accessor": "renewable",
                    },
                    Object {
                      "Header": "Service Type",
                      "accessor": "serviceType",
                    },
                    Object {
                      "Header": "Period",
                      "accessor": "period",
                    },
                    Object {
                      "Header": "Basic Service",
                      "accessor": "basicService",
                    },
                    Object {
                      "Header": "SKU OR Quick Code",
                      "accessor": "skuORQuickCode",
                    },
                    Object {
                      "Header": "DMA Name",
                      "accessor": "dmaName",
                    },
                    Object {
                      "Header": "Retail Price",
                      "accessor": "retailPrice",
                    },
                    Object {
                      "Header": "Duration",
                      "accessor": "duration",
                    },
                    Object {
                      "Header": "Display Order",
                      "accessor": "displayOrder",
                    },
                    Object {
                      "Header": "is Ads Enabled",
                      "accessor": "isAdsEnabled",
                    },
                    Object {
                      "Header": "Display Name",
                      "accessor": "displayName",
                    },
                    Object {
                      "Header": "Product Name",
                      "accessor": "productName",
                    },
                    Object {
                      "Cell": [Function],
                      "Header": "Actions",
                      "accessor": "[row identifier to be passed to button]",
                      "filterable": false,
                    },
                  ]
                }
                data={Array []}
                loading={false}
              />,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": "A",
              "nodeType": "class",
              "props": Object {
                "columns": Array [
                  Object {
                    "Header": "Currency Code",
                    "accessor": "currencyCode",
                  },
                  Object {
                    "Header": "Product Description",
                    "accessor": "productDescription",
                  },
                  Object {
                    "Header": "Product Category",
                    "accessor": "productCategory",
                  },
                  Object {
                    "Header": "Channel Partner Description",
                    "accessor": "channelPartnerDescription",
                  },
                  Object {
                    "Header": "Renewable",
                    "accessor": "renewable",
                  },
                  Object {
                    "Header": "Service Type",
                    "accessor": "serviceType",
                  },
                  Object {
                    "Header": "Period",
                    "accessor": "period",
                  },
                  Object {
                    "Header": "Basic Service",
                    "accessor": "basicService",
                  },
                  Object {
                    "Header": "SKU OR Quick Code",
                    "accessor": "skuORQuickCode",
                  },
                  Object {
                    "Header": "DMA Name",
                    "accessor": "dmaName",
                  },
                  Object {
                    "Header": "Retail Price",
                    "accessor": "retailPrice",
                  },
                  Object {
                    "Header": "Duration",
                    "accessor": "duration",
                  },
                  Object {
                    "Header": "Display Order",
                    "accessor": "displayOrder",
                  },
                  Object {
                    "Header": "is Ads Enabled",
                    "accessor": "isAdsEnabled",
                  },
                  Object {
                    "Header": "Display Name",
                    "accessor": "displayName",
                  },
                  Object {
                    "Header": "Product Name",
                    "accessor": "productName",
                  },
                  Object {
                    "Cell": [Function],
                    "Header": "Actions",
                    "accessor": "[row identifier to be passed to button]",
                    "filterable": false,
                  },
                ],
                "data": Array [],
                "loading": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
